packages/network-controller/src/NetworkController.ts

```diff
@@ -913,7 +916,7 @@ export class NetworkController extends BaseController<
       }
     }
 
-    if (networkChanged) {
+    if (networkChanged || this.#destroyed) {
       // If the network has changed, then `lookupNetwork` either has been or is
       // in the process of being called, so we don't need to go further.
```

Comment:
Maybe we should update this comment:

``` suggestion
      // If the network has changed, then `lookupNetwork` either has been or is
      // in the process of being called, so we don't need to go further.
      // If the controller has been destroyed, this is a signal that the messenger
      // may have been cleared of event subscriptions, so we also don't want to go
      // further so that we don't get an error when attempting to unsubscribe.
```